### PR TITLE
DBZ-2000 Content based router SMT

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-2000

### CODE REVIEW

**Comment by jpechane** [comment]
@gunnarmorling Please disregard the Sanity check failure, github has/had an incident related to github actions.

File: debezium-core/src/test/java/io/debezium/transforms/RouterTest.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import io.debezium.DebeziumException;
+import io.debezium.data.Envelope;
+
+/**
+ * @author Jiri Pechanec
+ */
+public class RouterTest {
+
+    private static final String LANGUAGE = "language";
+    private static final String EXPRESSION = "topic";
```
Comment (gunnarmorling) [inline debezium-core/src/test/java/io/debezium/transforms/RouterTest.java:29]: How about naming it `topic.expression`? That'll be more in line with `topic.regex` as known from the existing logical table router.

Comment (jpechane) [reply debezium-core/src/test/java/io/debezium/transforms/RouterTest.java]: @gunnarmorling Renamed and docs added

Comment (gunnarmorling) [reply debezium-core/src/test/java/io/debezium/transforms/RouterTest.java]: Thx!

File: documentation/modules/ROOT/pages/configuration/content-based-routing.adoc [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+The SMT is a Java class that encodes the routing logic.
+This is a very powerful mechanism but has two drawbacks:
+
+* It is necessary to compile the transformation upfront and deploy it to Kafka Connect.
+* Every change needs code recompilation and redeployment, leading to inflexible operations.
+
+To solve this problem, Debezium comes with the Content-Based Router SMT.
+This SMT allows the operator to write an expression that is evaluated for each event and according to the result, it is either kept in the original topic or re-routed to another one.
+The current implementation supports two languages in which the expression can be written: Groovy and JavaScript.
```
Comment (gunnarmorling) [inline documentation/modules/ROOT/pages/configuration/content-based-routing.adoc:28]:
```suggestion
The current implementation supports any scripting language which integrates with https://jcp.org/en/jsr/detail?id=223[JSR 223] (Scripting for the Java(TM) Platform).
```

File: documentation/modules/ROOT/pages/configuration/content-based-routing.adoc [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+* `key` - a key of the message
+* `value` - a value of the message
+* `keySchema` - the schema of the message key
+* `valueSchema` - the schema of the message value
+
+The `key` and `value` are of type `org.apache.kafka.connect.data.Struct` and `keySchema` and `valueSchema` are variables of type `org.apache.kafka.connect.data.Schema`.
+The expression can invoke arbitrary methods on the variables and should evaluate into a boolean value that decides whether the message is removed `true` or kept.
+Expressions should be side-effect free, i.e. they should *not* modify the passed variables in any way.
+
+[NOTE]
```
Comment (gunnarmorling) [inline documentation/modules/ROOT/pages/configuration/content-based-routing.adoc:62]: Haha, now it's you who adds an empty note :)

File: documentation/modules/ROOT/pages/configuration/content-based-routing.adoc [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+====
+
+====
```
Comment (gunnarmorling) [inline documentation/modules/ROOT/pages/configuration/content-based-routing.adoc:65]:
```suggestion
```

File: documentation/modules/ROOT/pages/configuration/filtering.adoc [hunk 5]
```diff
@@ -25,7 +25,7 @@ This is a very powerful mechanism but has two drawbacks:
 
 To solve this problem, Debezium comes with the Filtering SMT.
 This SMT allows the operator to write an expression that is evaluated for each event and according to the result, it is either filtered out or passed on for further processing.
-The current implementation supports two languages in which the expression can be written: Groovy and JavaScript.
+The current implementation is tested with two languages in which the expression can be written: Groovy and JavaScript.
```
Comment (gunnarmorling) [inline documentation/modules/ROOT/pages/configuration/filtering.adoc:28]:
```suggestion
The current implementation supports any scripting language which integrates with https://jcp.org/en/jsr/detail?id=223[JSR 223] (Scripting for the Java(TM) Platform).
```

**Comment by jpechane** [comment]
@gunnarmorling Suggestions accepted

**Comment by gunnarmorling** [comment]
Applied. Ending the week on a high note :)

**Pull request merged**